tests/integration/ambient/registry_setup_test.go

```diff
@@ -35,11 +35,22 @@ const (
 )
 
 func testRegistrySetup(ctx resource.Context) (err error) {
-	registry, err = registryredirector.New(ctx, registryredirector.Config{
-		Cluster:        ctx.AllClusters().Default(),
-		TargetRegistry: "kind-registry:5000",
-		Scheme:         "http",
-	})
+	var config registryredirector.Config
+
+	isKind := ctx.Clusters().IsKindCluster()
+
+	// By default, for any platform, the test will pull the test image from public "gcr.io" registry.
+	// For "Kind" environment, it will pull images from the "kind-registry".
+	// For "Kind", this is due to DNS issues in IPv6 cluster.
```

Comment:
I don't understand what you mean in this comment. What is due to DNS issues?

Reply:
Hi @dhawton 

The "Kind" based environment has an issue, while being configured in IPv6 format, it fails to resolve and pull the wasm images.

This is explained here - https://github.com/istio/istio/pull/47978#issuecomment-1823155014
And was addresses here as a fix - https://github.com/istio/istio/pull/51778

Reply:
A similar PR test fix to the current one, was already done in the telemetry test suite, as I mentioned in the description.
https://github.com/istio/istio/pull/54348

Reply:
Can you clarify the comment? "For Kind, this is due to DNS issues in IPv6 cluster" is unclear what 'this' is or what it is referring to. Maybe also link to the relevant issue.

Reply:
@dhawton 

Done.